src/testers.jl

```diff
@@ -28,25 +28,28 @@ Call `FiniteDifferences.j′vp`, with the option to ignore certain `xs`.
 - `∂xs::Tuple`: Derivatives estimated by finite differencing.
 """
 function _make_fdm_call(fdm, f, ȳ, xs, ignores)
-    sig = Expr(:tuple)
-    call = Expr(:call, f)
-    newxs = Any[]
-    arginds = Int[]
-    i = 1
-    for (x, ignore) in zip(xs, ignores)
-        if ignore
-            push!(call.args, x)
-        else
-            push!(call.args, Symbol(:x, i))
-            push!(sig.args, Symbol(:x, i))
-            push!(newxs, x)
-            push!(arginds, i)
+    ignores = collect(ignores)
+    function f2(sigargs...)
+        callargs = Any[]
+        j = 1
+        for (i, (x, ignore)) in enumerate(zip(xs, ignores))
+            if ignore
+                push!(callargs, x)
+            else
+                push!(callargs, sigargs[j])
+                j += 1
+            end
         end
-        i += 1
+        @assert j == length(sigargs) + 1
```

Comment:
Not sure what others think of using `asserts` in production code? @oxinabox @nickrobinson251 @willtebbutt

Reply:
Several points:
-  Huge fan in general, they are like a comment that makes sure it says up to date
- Often better ot use the 2 arg form `@assert(cond, explain)`
- Huge anti-fan of using them when it is not a programmer mistake that makes them fail (not the case here). 
- Less of a fan for when it introduces changes outside of the assert line, which this one does , but the one below doesn't (but can be worth it)
4. *Not the case here) there is no way to disable them except monkey-patching `Base` (viable I guess), so often have to be cautious about 
5. This is not a production package, nothing here goes in any production system, this is a testing support package